### PR TITLE
[SDK/CLI] Upload instance_result.jsonl and flow_logs

### PR DESCRIPTION
**Pull request by 0mza987**
# Description

- Upload `flow_logs`, which is the detailed logs for each line run.
- Upload `instance_results.jsonl`, which is a summary of each line run. Cloud run has this to let UI know metrics like `failure rate`.
  ```jsonl
  {"line_number": 2, "status": "Completed", "inputs.url": "https://www.youtube.com/watch?v=o5ZQyXaAv1g", "inputs.line_number": 2, "category": "Movie", "evidence": "Text content"}
  {"line_number": 1, "status": "Completed", "inputs.url": "https://www.youtube.com/watch?v=o5ZQyXaAv1g", "inputs.line_number": 1, "category": "Movie", "evidence": "Text content"}
  {"line_number": 0, "status": "Completed", "inputs.url": "https://www.youtube.com/watch?v=o5ZQyXaAv1g", "inputs.line_number": 0, "category": "Movie", "evidence": "Text content"}
  ```

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_cli/_utils.py [hunk 1]
```diff
@@ -409,6 +409,51 @@ def _try_delete_existing_run_record(run_name: str):
...
+        000000000_000000000.jsonl
+        000000001_000000001.jsonl
+        000000002_000000002.jsonl
+
+        Get a list of dict like this:
+            {"line_number": 0, "status": "Completed", "inputs.name": "hod", "inputs.line_number": 2, "result": "res"}
+            ...
+            ...
+
+    Note that inputs keys are prefixed with 'inputs.', but outputs keys are not.
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_cli/_utils.py:429]: LOL

File: src/promptflow-devkit/promptflow/_cli/_utils.py [hunk 2]
```diff
@@ -409,6 +409,51 @@ def _try_delete_existing_run_record(run_name: str):
         pass
 
 
+def get_instance_results(path: Union[str, Path]) -> List[Dict]:
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_cli/_utils.py:412]: may add some test later, guard the result line content is expected

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow-devkit/promptflow/_sdk/_constants.py [hunk 3]
```diff
@@ -474,6 +474,8 @@ class Local2Cloud:
     ASSET_NAME_DEBUG_INFO = "debug_info"
     ASSET_NAME_FLOW_OUTPUTS = "flow_outputs"
     EXECUTION_LOG = "logs/azureml/executionlogs.txt"
+    # instance_results.jsonl contains the inputs and outputs of all lines
+    FLOW_INSTANCE_RESULTS_FILE_NAME = "instance_results.jsonl"
```
Comment (zhengfeiwang) [inline src/promptflow-devkit/promptflow/_sdk/_constants.py:478]: one silly questoin, what is instance results? may add to the pr desc.

Comment (0mza987) [reply src/promptflow-devkit/promptflow/_sdk/_constants.py:478]: updated in description

File: src/promptflow-devkit/promptflow/_cli/_utils.py (hunk 2 above)
Comment (0mza987) [reply src/promptflow-devkit/promptflow/_cli/_utils.py:412]: Good one, will add the test in next pr.

**Review by zhengfeiwang — APPROVED** [review]

**Pull request merged**